src/middlewares/paginate.middleware.ts

```diff
@@ -1,5 +1,6 @@
 import { NextFunction, Request, RequestHandler, Response } from 'express';
 import { Product } from '../models';
+import { PaginationQuery } from '../controllers/products.controller';
```

Comment:
It's not recommended to import anything from controller in any files (other than routers). So you might add a interfaces directory and add the interfaces files their, you can import it in the controllers or whenever you need it.